dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.c

```diff
@@ -34,6 +37,10 @@ RecycleBin5File_RecycleBinFile_QueryInterface(
         *ppvObject = &s->recycleBinFileImpl;
     else if (IsEqualIID(riid, &IID_IRecycleBinFile))
         *ppvObject = &s->recycleBinFileImpl;
+    else if (IsEqualIID(riid, &IID_IExtractIconA) || IsEqualIID(riid, &IID_IExtractIconW))
+    {
+        return SHCreateFileExtractIconW(s->FullName, 0, riid, ppvObject);
```

Comment:
At some point (when directories get properly handled) you might need, schematically, some
```c
DWORD dwAttributes = (/* something to determine whether this is a directory that returns TRUE */) ? FILE_ATTRIBUTE_DIRECTORY : FILE_ATTRIBUTE_NORMAL;
```
and then use `dwAttributes` in lieu of the 2nd parameter of `SHCreateFileExtractIconW`.

Reply:
ok, i'm going to work on this

Reply:
Done